### PR TITLE
COL-650 Screen hidden categories from daily/weekly emails

### PR DESCRIPTION
**Pull request by paulkerschen**
https://jira.ets.berkeley.edu/jira/browse/COL-650

### CODE REVIEW

File: node_modules/col-activities/tests/test-daily.js [hunk 1]
```diff
@@ -388,6 +389,64 @@ describe('Daily activity emails', function() {
...
+                    });
+                  });
+                });
+              });
+            });
+          });
+        });
+      });
+    });
+  });
```
Comment (johncrossman) [inline node_modules/col-activities/tests/test-daily.js]: You want indentation? We got indentation!!

File: node_modules/col-assets/lib/api.js [hunk 2]
```diff
@@ -1008,6 +1008,16 @@ var canDeleteAsset = function(user, asset) {
...
 };
 
+/**
+ * Returns true if the asset is not hidden, not deleted, and not associated with only hidden categories.
+ *
+ * @param   {Asset}        asset             The asset to check for deleted or hidden status
+ * @return  {boolean}                        True if deleted or hidden
+ */
+var isDeletedOrHidden = module.exports.isDeletedOrHidden = function(asset) {
+  return !asset.visible || !!asset.deleted_at || !(_.isEmpty(asset.categories) || _.find(asset.categories, 'visible'));
```
Comment (johncrossman) [inline node_modules/col-assets/lib/api.js]: @paulkerschen is my brain too slow on a Friday or does the comment above ("Returns true if... not hidden") not match the true logic?

Comment (paulkerschen) [inline node_modules/col-assets/lib/api.js]: @johncrossman Oh, you're quite right - that comment is secret evidence that the method was originally written the other way round.

File: node_modules/col-activities/lib/notifications/daily.js [hunk 3]
```diff
@@ -493,13 +494,35 @@ var getCourseAssetData = function(ctx, users, dateRange, callback) {
...
       // The comments
-      {'model': DB.Comment, 'required': true, 'where': {'created_at': {'gt': dateRange.start, 'lt': dateRange.end}}, 'include': [
-        {'model': DB.Comment, 'required': false, 'as': 'parent'}
-      ]}
+      {
+        'model': DB.Comment,
+        'required': true,
+        'where':
+        {
+          'created_at': {
```
Comment (johncrossman) [inline node_modules/col-activities/lib/notifications/daily.js]: @paulkerschen minor concern: do we have a varying code style above w.r.t opening curly brace? Do we prefer:
```
  'created_at':
  {
    ...
```

**Comment by johncrossman** [comment]
Looks good, @paulkerschen  One or two minor amendments needed.

File: node_modules/col-activities/lib/notifications/daily.js (hunk 3 above)
Comment (paulkerschen) [inline node_modules/col-activities/lib/notifications/daily.js]: @johncrossman Ah, no, I think we prefer `'where': {`. Will fix.

Comment (johncrossman) [inline node_modules/col-activities/lib/notifications/daily.js]: Thanks, @paulkerschen That was my hunch.

**Pull request merged**